src/arrayMethodJoin.js

```diff
@@ -5,8 +5,29 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    const copyArray = [...this];
+    let joinString = '';
+    let stringSeparator = '';
+
+    if (separator === undefined) {
+      stringSeparator = ',';
+    } else {
+      stringSeparator = '' + separator;
+    }
+
+    for (let i = 0; i < copyArray.length; i++) {
+      if (copyArray[i] === null || copyArray[i] === undefined) {
+        copyArray[i] = '';
```

Comment:
2. [CODE STYLE] - avoid creating needless copies of arrays, you can iterate existing array.

I suppose you don't need here a copy of the array.
You could just create a variable inside a loop for saving an element of the array.

```
for (let i = 0; i < this.length; i++) {
      let el = this[i]; 
      if (el === null || el === undefined) {
        el =  '';
        ```

Reply:
Thank you so match! It's useful advice)